acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/tls/ClientSideTlsAcceptanceTest.java

```diff
@@ -0,0 +1,196 @@
+import tech.pegasys.ethsigner.tests.dsl.node.NodeConfigurationBuilder;
+import tech.pegasys.ethsigner.tests.dsl.node.NodePorts;
+import tech.pegasys.ethsigner.tests.dsl.signer.Signer;
+import tech.pegasys.ethsigner.tests.dsl.signer.SignerConfigurationBuilder;
+import tech.pegasys.ethsigner.tests.dsl.tls.TlsCertificateDefinition;
+import tech.pegasys.ethsigner.tests.tls.support.BasicPkcsStoreConfig;
+import tech.pegasys.ethsigner.tests.tls.support.MockBalanceReporter;
+import tech.pegasys.ethsigner.tests.tls.support.TlsEnabledHttpServerFactory;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import io.vertx.core.http.HttpServer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.web3j.protocol.core.methods.request.Transaction;
+import org.web3j.protocol.exceptions.ClientConnectionException;
+import org.web3j.utils.Convert;
+import org.web3j.utils.Convert.Unit;
+
+class ClientSideTlsAcceptanceTest {
+
+  private TlsEnabledHttpServerFactory serverFactory;
+
+  @BeforeEach
+  void setup() {
+    serverFactory = new TlsEnabledHttpServerFactory();
+  }
+
+  @AfterEach
+  void cleanup() {
+    serverFactory.shutdown();
+  }
+
+  private Signer createAndStartSigner(
+      final TlsCertificateDefinition presentedCert,
+      final TlsCertificateDefinition expectedWeb3ProviderCert,
+      final int downstreamWeb3Port,
+      final int listenPort,
+      final Path workDir)
+      throws IOException {
+    final Signer signer =
+        createSigner(
+            presentedCert, expectedWeb3ProviderCert, downstreamWeb3Port, listenPort, workDir);
+    signer.start();
+    signer.awaitStartupCompletion();
+
+    return signer;
+  }
+
+  private Signer createSigner(
+      final TlsCertificateDefinition presentedCert,
+      final TlsCertificateDefinition expectedWeb3ProviderCert,
+      final int downstreamWeb3Port,
+      final int listenPort,
+      final Path workDir)
+      throws IOException {
+
+    final Path clientPasswordFile =
+        Files.writeString(workDir.resolve("clientKeystorePassword"), presentedCert.getPassword());
+    final Path serverPasswordFile =
+        Files.writeString(
+            workDir.resolve("clientTrustStorePassword"), expectedWeb3ProviderCert.getPassword());
+
+    final SignerConfigurationBuilder builder = new SignerConfigurationBuilder();
+    builder.withDownstreamTrustStore(
+        new BasicPkcsStoreConfig(
+            expectedWeb3ProviderCert.getPkcs12File(), serverPasswordFile.toFile()));
+    builder.withDownstreamKeyStore(
+        new BasicPkcsStoreConfig(presentedCert.getPkcs12File(), clientPasswordFile.toFile()));
+    builder.withHttpRpcPort(listenPort);
+
+    final NodeConfiguration nodeConfig = new NodeConfigurationBuilder().build();
+    final NodePorts nodePorts = new NodePorts(downstreamWeb3Port, 9);
```

Comment:
what does 9 represents here?

Reply:
its unused across AT, is a bit of a YAGNI, but don't want to fix it here.
have commented its not used.

Reply:
looking at the code, its wsRPCPort. You can create a constant with UNUSED_WS_PORT=0 or similar. 9 looks like magic number :)